pallets/escrow/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+#![allow(
+    clippy::unused_unit,
+    unused_qualifications,
+    missing_debug_implementations
+)]
+#![cfg_attr(not(feature = "std"), no_std)]
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{dispatch::DispatchResultWithPostInfo, pallet_prelude::*};
+    use frame_system::pallet_prelude::*;
+    use orml_traits::{MultiCurrency, MultiReservableCurrency};
+    use vln_primitives::{EscrowDetail, EscrowHandler, EscrowId, EscrowState};
+
+    type BalanceOf<T> =
+        <<T as Config>::Asset as MultiCurrency<<T as frame_system::Config>::AccountId>>::Balance;
+    type AssetIdOf<T> =
+        <<T as Config>::Asset as MultiCurrency<<T as frame_system::Config>::AccountId>>::CurrencyId;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Because this pallet emits events, it depends on the runtime's definition of an event.
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// the type of assets this pallet can hold in escrow
+        type Asset: MultiReservableCurrency<Self::AccountId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::storage]
+    #[pallet::getter(fn rates)]
+    /// Escrows created by a user, this method of storageDoubleMap is chosen since there is no usecase for
+    /// listing escrows by provider/currency. The escrow will only be referenced by the creator in
+    /// any transaction of interest.
+    pub(super) type Escrow<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        T::AccountId,
+        Twox64Concat,
+        EscrowId,
```

Comment:
I think we should enforce people having only one escrow with any other user at a time, I don't see situations where a user might have more than one open trade with the same person(they can cancel and recreate the escrow with a different amount) in turn this restriction can help with automation and mitigate the cases where the user has more than one transfer in their account from the same recipient with the same amount.
All that just to say that the EscrowId can be just the recipient Account ID, so releasing an escrow for example just needs to know the recipient account.